bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,559 @@
+            deployment_spec.aws_ec2_operator_config.region = (
+                deployment_spec.aws_ec2_operator_config.region
+                or get_default_aws_region()
+            )
+            if not deployment_spec.aws_ec2_operator_config.region:
+                raise InvalidArgument("AWS region is missing")
+
+            ensure_sam_available_or_raise()
+            ensure_docker_available_or_raise()
+
+            bento_pb = self.yatai_service.GetBento(
+                GetBentoRequest(
+                    bento_name=deployment_spec.bento_name,
+                    bento_version=deployment_spec.bento_version,
+                )
+            )
+
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
+                raise BentoMLException(
+                    "BentoML currently not support {} repository".format(
+                        BentoUri.StorageType.Name(bento_pb.bento.uri.type)
+                    )
+                )
+            bento_path = bento_pb.bento.uri.uri
+
+            return self._add(deployment_pb, bento_pb, bento_path)
+        except BentoMLException as error:
+            deployment_pb.state.state = DeploymentState.ERROR
+            deployment_pb.state.error_message = f"Error: {str(error)}"
+            return ApplyDeploymentResponse(
+                status=error.status_proto, deployment=deployment_pb
+            )
+
+    def _add(self, deployment_pb, bento_pb, bento_path):
+        try:
+            if loader._is_remote_path(bento_path):
+                with loader._resolve_remote_bundle_path(bento_path) as local_path:
+                    return self._add(deployment_pb, bento_pb, local_path)
+
+            deployment_spec = deployment_pb.spec
+            aws_ec2_deployment_config = deployment_spec.aws_ec2_operator_config
+
+            artifact_s3_bucket_name = generate_aws_compatible_string(
+                "btml-{namespace}-{name}-{random_string}".format(
+                    namespace=deployment_pb.namespace,
+                    name=deployment_pb.name,
+                    random_string=uuid4().hex[:6].lower(),
+                )
+            )
+            create_s3_bucket_if_not_exists(
+                artifact_s3_bucket_name, aws_ec2_deployment_config.region
+            )
+
+            self.deploy_service(
+                deployment_pb,
+                deployment_spec,
+                bento_path,
+                aws_ec2_deployment_config,
+                artifact_s3_bucket_name,
+                aws_ec2_deployment_config.region,
+            )
+        except BentoMLException as error:
+            if artifact_s3_bucket_name and aws_ec2_deployment_config.region:
+                cleanup_s3_bucket_if_exist(
+                    artifact_s3_bucket_name, aws_ec2_deployment_config.region
+                )
+            raise error
+        return ApplyDeploymentResponse(status=Status.OK(), deployment=deployment_pb)
+
+    def delete(self, deployment_pb):
```

Comment:
I am not able to delete my test deployment with the following error:
```
[2020-10-14 00:25:57,648] ERROR - RPC ERROR DeleteDeployment: An error occurred (RepositoryNotEmptyException) when calling the DeleteRepository operation: The repository with name 'btml-repo-dev-test-my' in registry with id '192023623294' cannot be deleted because it still contains images
```

We probably want to remove the image from ECR and then delete the stack